appcues/src/main/java/com/appcues/session/DefaultSessionService.kt

```diff
@@ -0,0 +1,187 @@
+package com.appcues.session
+
+import android.os.Build.VERSION
+import com.appcues.AppcuesConfig
+import com.appcues.BuildConfig
+import com.appcues.R
+import com.appcues.analytics.AnalyticsEvent
+import com.appcues.analytics.AnalyticsIntent
+import com.appcues.analytics.AnalyticsIntent.Anonymous
+import com.appcues.analytics.AnalyticsIntent.Event
+import com.appcues.analytics.AnalyticsIntent.Identify
+import com.appcues.analytics.AnalyticsIntent.UpdateGroup
+import com.appcues.analytics.AnalyticsIntent.UpdateProfile
+import com.appcues.analytics.SessionRandomizer
+import com.appcues.analytics.SessionService
+import com.appcues.data.session.PrefSessionLocalSource
+import com.appcues.util.ContextResources
+import java.util.Date
+import java.util.UUID
+import java.util.concurrent.TimeUnit
+
+internal class DefaultSessionService(
+    private val config: AppcuesConfig,
+    private val sessionLocalSource: PrefSessionLocalSource,
+    private val contextResources: ContextResources,
+    private val sessionRandomizer: SessionRandomizer,
+) : SessionService {
```

Comment:
this is effectively merging the session management with the auto property decoration, is that accurate? If so, I think it might be nice still to separate those responsibilities.

Reply:
this class is a combination of SessionMonitor and AutoPropertyDecorator, yes. I think the sessionProperty has room here, the applicationProperties and autoProperties could be separate in a way, how would you do that?

Reply:
broke this down into two classes again, thanks for the feedback